### ISSUE TITLE
Changing the tolerance to alter EDM goal

### ISSUE THREAD

**Issue by mayou36** [scikit-hep/iminuit#353]
Hey, is there a possibility to change the tolerance of migrad? In order to alter the edm goal, which is composed of 0.001 * tolerance * UP.

The docs did not state anything AFAIK.
Many thanks!

**Comment by HDembinski** [comment]
https://iminuit.readthedocs.io/en/latest/reference.html

You can use "tol" (search for "Tolerance"). Let me know if it works.

**Comment by mayou36** [comment]
Didn't realize that it's a setter as well, sorry! But to confirm, it's not available as an argument to the constructor or as an argument to the method, right?

**Comment by HDembinski** [comment]
The constructor of Minuit is fully documented, see
https://iminuit.readthedocs.io/en/latest/reference.html
and especially the part "builtin keywords".
We could add a tolerance keyword if people want that.

**Comment by mayou36** [comment]
Thanks so far, I thought I've mixed something up somewhere. But then here the discrepancy that I see:

Setting the `tol` on the minimizer prints along the lines:
`start iterating until Edm is < 0.001`
but when it finds a minimum (as an example)
`EDM = 0.000708 (Goal: 5E-05)`
where the EDM seems realistic, but the _printed_ goal is not changed.

So the print of the Goal in the summary seems wrong.
tested with 1.3.7

**Comment by HDembinski** [comment]
Thank you for reporting. Unfortunately, often bugs like these are actually in MINUIT2 and not in iminuit, so we cannot really do much about it other than reporting it upstream to the ROOT devs.

You can see here how `tol` is handled: https://github.com/scikit-hep/iminuit/blob/develop/iminuit/_libiminuit.pyx, just search for `self.tol`. The `call_mnapplication_wrapper` is here https://github.com/scikit-hep/iminuit/blob/develop/iminuit/Utils.h. `cfmin2struct` is here https://github.com/scikit-hep/iminuit/blob/develop/iminuit/Minuit2Struct.pxi. It looks to me like we pass the tolerance correctly to MINUIT2 and get the EDM back from MINUIT2 FunctionMinimum object. The goal is computed here:
https://github.com/scikit-hep/iminuit/blob/develop/iminuit/repr_text.py, line 30. The goal is not the tolerance.
It looks to me that

**Comment by mayou36** [comment]
So I've checked it out:

- UP seems to be taken into account correctly
- the "start iterating until Edm < something" is the edmval multiplied by 0.002, I assume it's the 0.001 (as written in the docs) times a factor of two for the F77 compatibility.
- Where the EDM for iminuit is calculated, are you sure about the 1E-4 for the value and that it deviates from the docs? https://github.com/scikit-hep/iminuit/blob/18ea7b2c94b87d2c47a8d18d4ba21c41d2e1c804/iminuit/repr_text.py#L30? because if that is set to 1E-3, things would be correct. (except of that factor of two between the "start iterating until..." and the goal edm. Probably also take that into account in the goal edm?)

Unfortunately, the link to the code which seemingly has the 1E-4 factor that deviates from the docs is broken.

**Comment by HDembinski** [comment]
Right, I found the same. I think this is indeed a bug in the formula that we use. I already have a feature branch to update this. Thank you again for reporting and investigating. :)

**Comment by HDembinski** [comment]
I am not sure whether this is fixed now, is there an easy way to check?

**Comment by mayou36** [comment]
Sorry, I missed that post. I usually just run a fit and do visual inspection of the output, no other way.

You may wanna push the branch and make a (WIP) PR?

**Comment by HDembinski** [comment]
This is fixed in the develop branch now.

**Comment by alexander-held** [comment]
Hi, I came across this while trying to understand a discrepancy between `start iterating until Edm is ...` and the EDM goal shown in the summary table, which was caused by me using an `iminuit` version from before the fix. With 1.4.9 now the values match.

While debugging, I noticed that the [documentation for tolerance](https://iminuit.readthedocs.io/en/latest/reference.html#iminuit.Minuit.tol) shows the calculation

```
edm_max = 0.002 * tol * errordef
```

which will default to 2e-4 and matches my printouts. [This section](https://iminuit.readthedocs.io/en/latest/reference.html#function-minimum-data-object) shows the criterion for `is_above_max_edm` as

```
Is EDM above 0.0001*tolerance*up?
```

Should the number here also be 0.002, or is this an unrelated calculation? The [errordef section](https://iminuit.readthedocs.io/en/latest/reference.html#iminuit.Minuit.errordef) explains that `errordef` = `up`.
